src/Host/Client/Impl/Extensions/RStringExtensions.cs

```diff
@@ -17,7 +17,18 @@ public static class RStringExtensions {
                 return nullValue;
             }
 
-            return quote + s.Replace("\\", "\\\\").Replace("" + quote, "\\" + quote) + quote;
+            return quote + s.Replace("\\", "\\\\").Replace("" + quote, "\\" + quote).EscapeCharacterConstants() + quote;
+        }
+
+        private static string EscapeCharacterConstants(this string s) {
+            s = s.Replace("\n", "\\n")
```

Comment:
To clarify, when I wrote `\\n` above, I meant those literal three characters typed by the user - not `\n` as in newline. In C# terms, `@"\\n"`. So far as I can see, the rest of the reasoning is valid if you start from that point.

If the user actually types a backslash followed by the newline, we need to convert the newline itself to `\n` - there's no dealing with backslashes in the input in that case, so it's a different thing.